server/src/main/java/org/apache/iotdb/db/auth/ClusterAuthorityFetcher.java

```diff
@@ -0,0 +1,337 @@
+      return status.getStatus();
+    } else {
+      return status.getStatus();
+    }
+  }
+
+  /**
+   * Initialize user and role cache information.
+   *
+   * <p>If the permission information of the role changes, only the role cache information is
+   * cleared. During permission checking, if the role belongs to a user, the user will be
+   * initialized.
+   */
+  public boolean invalidateCache(String username, String roleName) {
+    if (userCache.getIfPresent(username) != null) {
+      List<String> roleList = userCache.getIfPresent(username).getRoleList();
+      if (!roleList.isEmpty()) {
+        roleCache.invalidateAll(roleList);
+      }
+      userCache.invalidate(username);
+    }
+    if (roleCache.getIfPresent(roleName) != null) {
+      roleCache.invalidate(roleName);
+    }
+    if (userCache.getIfPresent(username) != null && roleCache.getIfPresent(roleName) != null) {
+      logger.error("datanode cache initialization failed");
+      return false;
+    }
+    return true;
+  }
+
+  /** cache user */
+  public User cacheUser(TPermissionInfoResp tPermissionInfoResp) {
+    User user = new User();
+    List<String> privilegeList = tPermissionInfoResp.getUserInfo().getPrivilegeList();
+    List<PathPrivilege> pathPrivilegeList = new ArrayList<>();
+    user.setName(tPermissionInfoResp.getUserInfo().getUsername());
+    user.setPassword(tPermissionInfoResp.getUserInfo().getPassword());
+    for (int i = 0; i < privilegeList.size(); i++) {
+      String path = privilegeList.get(i);
+      String privilege = privilegeList.get(++i);
+      pathPrivilegeList.add(toPathPrivilege(path, privilege));
+    }
+    user.setPrivilegeList(pathPrivilegeList);
+    user.setRoleList(tPermissionInfoResp.getUserInfo().getRoleList());
+    for (String roleName : tPermissionInfoResp.getRoleInfo().keySet()) {
+      roleCache.put(roleName, cacheRole(roleName, tPermissionInfoResp));
+    }
+    return user;
+  }
+
+  /** cache role */
+  public Role cacheRole(String roleName, TPermissionInfoResp tPermissionInfoResp) {
+    Role role = new Role();
+    List<String> privilegeList = tPermissionInfoResp.getRoleInfo().get(roleName).getPrivilegeList();
+    List<PathPrivilege> pathPrivilegeList = new ArrayList<>();
+    role.setName(tPermissionInfoResp.getRoleInfo().get(roleName).getRoleName());
+    for (int i = 0; i < privilegeList.size(); i++) {
+      String path = privilegeList.get(i);
+      String privilege = privilegeList.get(++i);
+      pathPrivilegeList.add(toPathPrivilege(path, privilege));
+    }
+    role.setPrivilegeList(pathPrivilegeList);
+    return role;
+  }
+
+  /**
+   * Convert user privilege information obtained from confignode to PathPrivilege
+   *
+   * @param path permission path
+   * @param privilege privilegeIds
+   * @return
+   */
+  private PathPrivilege toPathPrivilege(String path, String privilege) {
+    PathPrivilege pathPrivilege = new PathPrivilege();
+    String[] privileges = privilege.replace(" ", "").split(",");
```

Comment:
can use `privilege.trim()` to replase `privilege.replace(" ", "")`

Reply:
trim() can only delete the leading and trailing spaces. Here,  need to remove the middle spaces as well